jwst/outlier_detection/utils.py

```diff
@@ -75,6 +77,16 @@ def create_median(resampled_models, maskpt, on_disk=True, buffer_size=10.0):
         return np.nanmedian(np.array(model_list), axis=0)
     else:
         # set up buffered access to all input models
+        log.info("Computing median in chunks to save memory")
+        if allowed_memory is None:
+            allowed_memory = 0.5
+        machine_available_memory = psutil.virtual_memory().available + psutil.swap_memory().total
```

Comment:
I'm not sure why swap memory is included. Unless I misunderstand what this refers to (and I may very well misunderstand), I don't think the size of the swap area should be included. We are most interested in how much real memory is available so no paging or swapping is necessary.

Reply:
Thanks for reviewing this.  Brett pointed out the same thing, and it's buried somewhere in our long back-and-forth (understandable that you did not read it all lol).  If the swap memory were removed, what is your take on whether it's better to use the total amount of available memory or a per-model section size?

Reply:
Yes, I just noticed the long discussion "below the fold". I had some general concerns as well along Brett's lines.